src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <tuple>
+#include <unordered_map>
+#include <variant>
+
+#include "logging.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "util.hpp"
+
+namespace ov {
+namespace npuw {
+namespace weights {
+
+enum class TransformType : int { TENSOR, PERMUTE, CONVERT, CONCAT };
+
+// Forward declaration
+class LazyTensor;
+struct LazyTensorImpl;
+
+using ConcatMeta = std::pair<std::vector<LazyTensor>, std::size_t>;
+using ConstPtr = std::shared_ptr<ov::op::v0::Constant>;
+using LTData = std::variant<ConstPtr, ov::Tensor>;
+
+// LazyTensor owns Constant's memory
+using Transform = std::variant<LTData, std::vector<std::size_t>, std::monostate, ConcatMeta>;
```

Comment:
not sure how the comment is relevant to the below line

Reply:
Done